app/services/course/duplication/object_duplication_service.rb

```diff
@@ -30,23 +32,38 @@ def time_shift(current_course, target_course)
   end
 
   DEFAULT_OBJECT_DUPLICATION_OPTIONS =
-    { mode: :object, current_user: User.system }.freeze
+    { mode: :object, unpublish_all: true, current_user: User.system }.freeze
 
   # Duplicate the objects with the duplicator.
   #
-  # @return [Course] The duplicated objects
+  # @param [Object|Array] objects An object or an array of objects to duplicate.
+  # @return [Object] The duplicated object, if `objects` is a single object.
+  # @return [Array] Array of duplicated objects, if `objects` is an array.
   def duplicate_objects(objects)
-    # TODO: Inform the user when the duplication is complete.
+    # TODO: Email the user when the duplication is complete.
     Course.transaction do
       duplicated = duplicator.duplicate(objects)
-      success = duplicated.respond_to?(:save) ? duplicated.save : duplicated.all?(&:save)
-      raise ActiveRecord::Rollback unless success
+      save_success = duplicated.respond_to?(:save) ? duplicated.save : duplicated.all?(&:save)
+      after_save_success = save_success && after_save(objects, duplicated)
+      raise ActiveRecord::Rollback unless after_save_success
       duplicated
     end
   end
 
   private
 
+  # Executes callbacks meant to be invoked after duplicated objects have been saved
```

Comment:
Can I trouble you to add some documentation to the top of this service describing about how models can define such callbacks?

Reply:
Added!